### PR TITLE
[993] deserialize session info

### PR DESCRIPTION
**Pull request by defong**
### Context
Changes introduced by frontend

### Changes proposed in this pull request
Accommodating the frontend changes

### Guidance to review
Introduction of ` deserializable_resource :session`

Notable context
`"no params"`
`"invalid resource type"`

### CODE REVIEW

File: spec/requests/api/v2/session_spec.rb [hunk 1]
```diff
@@ -138,29 +136,39 @@
...
-    it 'raise errors' do
+    it 'does not update user record' do
       expect {
-        post '/api/v2/sessions',
-          headers: { 'HTTP_AUTHORIZATION' => credentials }
-      }.to raise_error NoMethodError
+        post(
+          '/api/v2/sessions',
+          headers: { 'HTTP_AUTHORIZATION' => credentials },
+          params: params
```
Comment (defong) [inline spec/requests/api/v2/session_spec.rb]: you broken the context  `no params`

File: app/controllers/concerns/validate_jsonapi_type.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  def validate_jsonapi_type(params, type)
+    # jsonapi-rb doesn't appear to validate the type of data record that's
+    # been given us, so even though we say "require(:session)" it won't
+    # validate the type field of the data object that was passed through.
+    #
+    # So for now, we do this by hand.
+    sent_type = params[:_jsonapi][:data][:type]
+    unless sent_type == type
+      raise ActionController::BadRequest.new(
+        "data type '#{sent_type}' unsupported'"
```
Comment (defong) [inline app/controllers/concerns/validate_jsonapi_type.rb]: may add expected type as well, thinking of when we need to scan thro error logs

Comment (defong) [reply app/controllers/concerns/validate_jsonapi_type.rb]: `"data type expected #{type} but received data type '#{sent_type}' unsupported'"`

File: spec/requests/api/v2/session_spec.rb (hunk 1 above)
Comment (misaka) [reply spec/requests/api/v2/session_spec.rb]: Ugh. You're right. I spotted that earlier too and thought I had fixed it.

File: app/controllers/concerns/validate_jsonapi_type.rb (hunk 2 above)
Comment (misaka) [reply app/controllers/concerns/validate_jsonapi_type.rb]: Sounds good!

**Comment by misaka** [comment]
@defong I think this is ready to be brought out of draft now.

**Comment by defong** [comment]
[Create new user sessions in API backend](https://trello.com/c/gHNepiOF/993-create-new-user-sessions-in-api-backend)

**Pull request merged**